src/template/wasm2c_atomicops.declarations.c

```diff
@@ -0,0 +1,114 @@
+
+#define DEFINE_ATOMIC_LOAD(name, t1, t2, t3)                            \
+  static inline t3 name(wasm_rt_memory_t* mem, u64 addr) {              \
+    MEMCHECK(mem, addr, t1);                                            \
+    t1 result;                                                          \
+    wasm_rt_memcpy(&result, &mem->data[addr], sizeof(t1));              \
+    result = __atomic_load_n((t1*) &mem->data[addr], __ATOMIC_SEQ_CST); \
```

Comment:
I suppose the alternative would be to use C11 atomics (https://en.cppreference.com/w/c/atomic).  Those might be portable to MSVC?  But then would be increasing the minimum C version from C99 for C11.. maybe this is OK if its only for modules that need threading support?

Reply:
I think I would be ok with that approach as well. My thinking is for now, we can stick with the current approach as it supports older compilers reasonably also. Support for msvc using msvc intrinsics doesn't look too bad either. If for whatever reason, msvc intrinsics become hard to incorporate, we can just use c11 atomics.

Reply:
Also looks like msvc got support for c11 atomics fairly recently. So we probably want to go through the intrinsic route if we want to support some of the older msvc compilers.

https://devblogs.microsoft.com/cppblog/c11-atomics-in-visual-studio-2022-version-17-5-preview-2/

Reply:
I did play with C atomics as well. Unfortunately, we can't use the C atomic ops as

- c atomic ops only works on atomic types `_Atomic`
- casting to/from atomics from/to a regular pointer in the Wasm heap is undefined behavior in C https://stackoverflow.com/questions/55299525/casting-pointers-to-atomic-pointers-and-atomic-sizes

Reply:
Latest PR implements MSVC instrincs as well